tests/OmniSharp.Roslyn.CSharp.Tests/SignatureHelpFacts.cs

```diff
@@ -474,38 +474,294 @@ public Program(Program p, int n)
     {
     }
 }";
-
             var actual = await GetSignatureHelp(source);
             Assert.Equal(3, actual.Signatures.Count());
             Assert.Equal(1, actual.ActiveParameter);
             Assert.Contains("ctor2", actual.Signatures.ElementAt(actual.ActiveSignature).Documentation);
         }
 
         [Fact]
-        public async Task SkipReceiverOfExtensionMethods()
+        public async Task TestForInheritedMethods()
         {
             const string source =
-@"class Program
+@"public class MyBase
 {
-    public static void Main()
+    public void MyMethod(int a) { }
+    public void MyMethod(int a, int b) { }
+}
+
+public class Class1 : MyBase
+{
+    public void MyMethod(int a, int b, int c) { }
+    public void MyMethod(int a, int b, int c, int d) { }
+}
+
+public class Class2
+{
+    public void foo()
     {
-        new Program().B($$);
+        Class1 c1 = new Class1();
+        c1.MyMethod($$);
     }
-    public Program()
+
+ }";
+            var actual = await GetSignatureHelp(source);
+            Assert.Equal(4, actual.Signatures.Count());
+        }
+
+        [Fact]
+        public async Task InheritedInaccesibleMethods()
+        {
+            const string source =
+@"public class MyBase
+{
+    private void MyMethod(int a) { }
+}
+
+public class Class1 : MyBase
+{
+    public void MyMethod(int a, int b, int c) { }
+    protected void MyMethod(int a, int b, int c, int d) { }
+}
+
+public class Class2
+{
+    public void foo()
     {
+        Class1 c1 = new Class1();
+        c1.MyMethod($$);
     }
-    public bool B(this Program p, int n)
+
+ }";
+            var actual = await GetSignatureHelp(source);
+            Assert.Single(actual.Signatures);
```

Comment:
Should we verify which signature we got?

Reply:
Yes we should.Added that.